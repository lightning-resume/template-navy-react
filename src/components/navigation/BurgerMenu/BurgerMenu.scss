@import 'src/styles/colors.scss';

.burger-container {
  transform: translateZ(0);
  transition: transform 0.3s ease-in-out;
  &.active {
    transform: rotateZ(45deg);

    & .first {
      transform: translateY(8px);
    }
    & .second {
      transform: rotateZ(90deg);
    }
    & .third {
      opacity: 0;
    }
  }
}
.burger-line {
  height: 3px;
  width: 20px;
  background-color: $lightest-slate;
  margin: 5px 0;
  transition: transform 0.2s ease-in-out;
}
