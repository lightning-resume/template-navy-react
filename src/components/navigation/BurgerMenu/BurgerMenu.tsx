import './BurgerMenu.scss'

export const BurgerMenu = ({ isActive, onClick }: { isActive?: boolean; onClick: () => void }) => {
  return (
    <div
      aria-label="navigation menu"
      onClick={onClick}
      className={['burger-container', isActive ? 'active' : 'no-active'].join(' ')}
    >
      <div className="burger-line first" />
      <div className="burger-line second" />
      <div className="burger-line third" />
    </div>
  )
}
