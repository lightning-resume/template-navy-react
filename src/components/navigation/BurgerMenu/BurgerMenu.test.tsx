import { render, screen, fireEvent } from '@testing-library/react'

import { BurgerMenu } from './BurgerMenu'

describe('BurgerMenu', () => {
  const mockOnClick = jest.fn()
  afterEach(() => {
    jest.clearAllMocks()
  })
  describe('when no active', () => {
    it('should render the component', () => {
      const { container } = render(<BurgerMenu onClick={mockOnClick} />)

      expect(container).toMatchSnapshot()
    })
  })
  describe('when active', () => {
    it('should render the component', () => {
      const { container } = render(<BurgerMenu onClick={mockOnClick} isActive={true} />)

      expect(container).toMatchSnapshot()
    })
  })
  describe('when clock on the burger', () => {
    it('should call callback when click on burgermenu', () => {
      render(<BurgerMenu onClick={mockOnClick} isActive={true} />)
      fireEvent.click(screen.getByLabelText('navigation menu'))

      expect(mockOnClick).toHaveBeenCalledTimes(1)
    })
  })
})
