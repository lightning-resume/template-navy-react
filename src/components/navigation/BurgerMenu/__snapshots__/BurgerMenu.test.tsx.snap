// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BurgerMenu when active should render the component 1`] = `
<div>
  <div
    aria-label="navigation menu"
    class="burger-container active"
  >
    <div
      class="burger-line first"
    />
    <div
      class="burger-line second"
    />
    <div
      class="burger-line third"
    />
  </div>
</div>
`;

exports[`BurgerMenu when no active should render the component 1`] = `
<div>
  <div
    aria-label="navigation menu"
    class="burger-container no-active"
  >
    <div
      class="burger-line first"
    />
    <div
      class="burger-line second"
    />
    <div
      class="burger-line third"
    />
  </div>
</div>
`;
